Wird es eine Unterstützung für RockyLinux geben?

smartonaut opened the pull request:
Moin Moin, da RedHat ja nun nicht mehr ist würde ich mir eine Unterstützung für Rocky Linux Alma und Oracle wünschen, wäre das möglich?

WXbet committed:
Update libusb url detection

WXbet committed:
Fix zlib 1.12.2 build

WXbet committed:
s3.TUP crosstool-NG toolchain template updates

linux-5.16.9
binutils-2.38
gettext-0.21
mpc-1.2.1

WXbet committed:
s3.TUP: Fix PCSClite check (v0.23.6)

- add trailing slash on prefixdir variable if not exists

Commit:
s3.TUP: library updates

- fix detection of OpenSSL 3.0.x (SSL_300)
- add OpenSSL 3.1.x beta library (SSL_310)
- add cURL 7.x.x library (LIB_CURL)
- add option to configure authenticated curl requests (CURL_GITHUB_TOKEN)

Commit:
s3.TUP crosstool-NG toolchain template updates

linux-6.1
binutils-2.39
gcc-12.2.0
expat-2.5.0
zlib-1.2.13

Commit:
s3.TUP crosstool-NG toolchain template fix compile issues

- disable -fPIC which is enabled by default since gcc 11.3.0

Commit:
s3.TUP: Fix freetz-ng build (v0.23.7)

- improve target detection during toolchain build

Commit:
s3 2023.01 + s3.TUP (v0.24.0)

- integrate toolchain download in s3.TUP

Commit:
Fix base64 encoding for download url

Commit:
Add option to create static toolchains

- set simplebuild3 as CT_TOOLCHAIN_PKGVERSION string

Commit:
Fix libcurl detection

Commit:
Update pcsc-lite library

- add C11 compiler compatibility check for latest PCSC-Lite
- add PCSC-Lite 1.9.5 to use with older compiler
- remove CURL_GITHUB_TOKEN setting

Commit:
Change CURL_GITHUB_TOKEN setting

for development purposes

Commit:
Add download toolchain dcube

Commit:
Switch to build static toolchains

Commit:
Remove double popup menu during editing templates

- this also remove the function to 'save as' an edited template

Commit:
Replace toolchain dcube with oe20_armv7

Commit:
Fix stuck after pcsclite when integrating multiple libraries

Commit:
Generate file hashes when running in development mode

Commit:
Remove self_build marker from download toolchains

Commit:
Update toolchain template oe20_armv7

Commit:
Restore download toolchain dcube

WXbet committed:
new download location for legacy toolchains

WXbet committed:
Ready for s3 2023.01.315

Commit:
Auto-Detect patches (Normal and streamboard)

Normal patches (-p1) and streamboard-style patches (-p0) are now automatically detected and applied accordingly.

Co-Authored-By: Schimmelreiter <2022025+Schimmelreiter@users.noreply.github.com>
Co-Authored-By: cojo11 <5813057+cojo11@users.noreply.github.com>

Commit:
Fix permissions

WXbet committed:
synology_armv5: add extra_cc NO_PTHREAD_STACKSIZE

- this removes weired warning "pthread_attr_setstacksize() failed in start_thread with error 22 Invalid argument" from oscam log
- see https://board.streamboard.tv/forum/thread/48195-gel%C3%B6st-ds212plus-oscam-v11718-https-trac-streamboard-tv-oscam-ticket-4277/

WXbet committed:
OpenSSL: template updates

- version 3.1.0 isn't beta anymore
- fix version 1.0.0 download url

WXbet commented:
Was geht denn da nicht?

WXbet commented:
> Moin Moin, da RedHat ja nun nicht mehr ist würde ich mir eine Unterstützung für Rocky Linux Alma und Oracle wünschen, wäre das möglich?

Was geht denn da nicht?

Commit:
s3.TUP new crosstool-NG toolchain template owrt_mediatek_armv8

support for ARMv8 aarch64 (cortex-a53) devices like GLinet Brume2
musl 1.1.24
gcc 13.1.0
linux 6.3.2

Commit:
s3.TUP new crosstool-NG toolchain template owrt_rpi_armv6 + owrt_rpi_armv7

support for OpenWrt 19.07|21.02 Raspberry Pi Zero W|WH, 1 Model A|A+|B|B+ (armv6)
support for OpenWrt 19.07|21.02 Raspberry Pi 2 Model B (armv7)
musl 1.1.24
gcc 13.1.0
linux 6.3.2

Commit:
s3.TUP: improve automatic library version and url detection

Commit:
s3.TUP crosstool-NG toolchain template updates

OS              : linux-6.5.5
Binutils        : binutils-2.40
Compiler        : gcc-13.2.0
Companion libs  : cloog-0.18.4 expat-2.5.0 gettext-0.21 gmp-6.2.1 isl-0.26 libelf-0.8.13 libiconv-1.16 mpc-1.3.1 mpfr-4.2.1 ncurses-6.4 zlib-1.2.13 zstd-1.5.5

Commit:
s3.TUP Android NDK toolchain template updates

Android NDK r26 LTS
OS : linux-6.3.0
Compiler : clang-17.0.2

update build scripts to work with Android NDK r26 LTS: ANDK_BUILD, SSL_320, SSL_310, SSL_300, SSL_111, LIB_CCID, LIB_ZLIB

WXbet committed:
Update french translation

thanks sam_ from streamboard.tv

Commit:
Improve tccheck command output

- tccheck: consolidate output and additionally show toolchain's linux and libc version
- s3.tup: show Android NDK's linux version in corresponding toolchains, default value for endianess

Commit:
s3.TUP: Fix libusb 1.0.23 download url

Commit:
Further improvement of tccheck command output

- show pkgconfig libraries for native toolchain
- better printf output formatting
- s3.tup: update function call to get linux version number
- bump new release version 2023.11.319

Commit:
s3.TUP library config updates

- SSL_320: remove beta tag from OpenSSL 3.2 as it is final now
- LIB_PCSC: PCSClite 2.0.1 has enabled enapolkit by default, so disable-polkit was added to build it without polkit dependency

Commit:
Fix whitespace typos

Commit:
tccheck: improve glibc version detection

Commit:
s3.TUP: Add support for multilib detection

- Library Update Menu now correctly displays multilibs (e.g. OpenSSL)
- fix wrong compiler compatibility message
- minor formatting fix

Commit:
s3.TUP Freetz-NG toolchain template updates

- remove outdated cross toolchain templates (fos_g71x_mips, fos_m72x_mips, fos_mLAB_mips, fos_u69x_mips, fos_u71x_arm, fos_u71x_mips, fos_u71x_puma, fos_uLAB_mips)
- add new download toolchains (fritz3272_7272_7330_7330SL_7390_ssl_multi_mips-freetz, fritz3370_3390_6840_7360v2_ssl_multi_mips-freetz, fritz3490_5490_5491_7430_7490_ssl_multi_mips-freetz, fritz4020_ssl_multi_mips-freetz, fritz4040_4060_5590_68504g_68505g_7510_7520_7520B_7530_ssl_multi_arm-freetz, fritz6430_6490_6590_ssl_multi_i686-freetz, fritz6591_6660_6690_ssl_multi_i686-freetz, fritz6890_7560_7580_7583_7583vdsl_7590AX_7590_ssl_multi_mips-freetz, fritz7362_ssl_multi_mips-freetz, fritz7530AX_ssl_multi_arm-freetz) provided by MasterSaMMy aka SaMMyMaster aka MasterRoCcO

Co-Authored-By: MasterRoCcO <47879729+masterrocco@users.noreply.github.com>

Commit:
s3.TUP Add library static build option

Commit:
s3.TUP new crosstool-NG toolchain templates vuplus4k_armv7,vuplus4k_armv7a

- for Vu+ Zero|Uno|Duo|Solo|Ultimo 4K with improved stability
- arch cortex-a15 32-bit LE
- glibc 2.25
- gcc 13.2.0
- linux 6.6.1

Commit:
tccheck: add gcc c11 support detection

Commit:
s3.TUP static build fixes

fixes for libusb, libpcsclite and ccid

Commit:
Introduce USE_STATIC option

- option USE_STATIC is available for commandline + gui build, profiles in the same way as the existing ones
- static embed libcrypto, ssl, libusb and/or libpcsclite
- these libraries must exist statically in the cross toolchain to be built properly

Commit:
s3.TUP Freetz-NG toolchain template updates

freetz toolchains pack6

Co-Authored-By: MasterRoCcO <47879729+MasterRoCcO@users.noreply.github.com>

Commit:
tccheck: add static/dynamic library detection

- show zlib header files

Commit:
s3.TUP Do not delete the temp folder

in case of build errors

Commit:
Switch oscam-emu url to upstream repository

Commit:
Support for multiple patch urls

Commit:
Remove svn remnants

as github will sunset Subversion support on January 8, 2024
- see https://github.blog/2023-01-20-sunsetting-subversion-support/
- replace svn with git in gitrevision, update_me and update_emu functions
- remove svn part from s3url and emuurl variables

Commit:
Add neon to oscam_name

WXbet committed:
s3.TUP library config updates

- Prevent download urls from being created from alpha, beta, pre or rc release tags
- get assets browser_download_url dynamically by extension

Commit:
s3.TUP Check build system and cross toolchain host architecture

Toolchain menu: Displaying a warning message and highlight incompatible toolchains if the architecture of the build system and the host (not target!) architecture of the cross toolchain are different

Commit:
tccheck: add cross toolchain host/target architecture detection

Commit:
s3.TUP Android toolchain updates

- symlink the correct clang wrapper script to remove missing compiler C11 support warnings
- new cross toolchain templates for Android 12 + 13

Commit:
s3.TUP: fix curl configure error

disable libpsl dependency

Commit:
Add URL checks before performing online update actions

suggestion from @kitte888

Commit:
s3.TUP Remove svn remnants

and add URL check before performing online update actions

Commit:
Introduce EXTRA_CFLAGS

- define additional CFLAGS in toolchain.cfg (e.g. extra_c="-mfpu=vfpv3-d16"; #disable NEON optimization for a specific arm cpu)
- documentation: https://github.com/gorgone/s3_releases/wiki/Configuration-File-toolchain.cfg

Commit:
Fix neon in oscam_name

Commit:
Fix resetting all optional variables from toolchain.cfg

Commit:
Revert "Fix neon in oscam_name"

This reverts commit 284c21486f393a6496941682a7728770d94b42a7.

WXbet committed:
Update compiler flags

* Update compiler flags

* Add USE_EXTRA option
This is to enable the use of the extra_* variables in the toolchain.cfg files.
By default (without this option) these variables are now ignored.

Commit:
Reorder neon tag in oscam_name

Commit:
Introduce applied patches in oscam webif

this works only for patches applied via s3 gui, cmd, profile

Commit:
Add icam to oscam_name

Commit:
Fix neon in oscam_name

Commit:
Improve applied patches in oscam webif

- now it is possible to write patch information to the SVN-IS-PATCHED file from outside without using the patching functionality of s3
- the format should be (on one line): patchname1.patch\n\t\tpatchname2.patch\n\t\tpatchname3.patch
- the content of SVN-IS-PATCHED file is integrated into WebIf during the s3 build process

Commit:
Fix patching during build on commandline

- USE_PATCH: patch as early as possible to incorporate and take advantage of potentially new build options that the automatically applied patches bring in

Commit:
Update compiler flags

- remove -fno-tree-vectorize as it produces problem on some devices (vuultimo4k, dreambox)
- so be more compatible by default

Commit:
Fix neon in oscam_name

- support aarch64

Commit:
Update compiler flags

- remove -funroll-loops as it produces problems on older toolchains (e.q. arm_dream)
- this reduzes the size of the resulting binary
- so be more compatible by default

Commit:
Delete no.patch file on startup

- this is a dummy file for demonstrattion in support\patches\patch.url.template

Commit:
Support libdvbcsa

- support for integration of libdvbcsa library into cross toolchain
- support for static linking of libdvbcsa library into oscam binary

Commit:
Make sure directory pkgconfig exists

Commit:
Fix libdvbcsa static build support

Commit:
Support libdvbcsa benchmark tools

- creating a zipfile containing libdvbcsa benchmark tools
- creating a zipfile containing libdvbcsa libraries tools

Commit:
Disable libdvbcsa benchmark tools by default

WXbet committed:
Ready for s3 2024.03.332

WXbet committed:
Ready for s3.TUP (0.25.0)

Commit:
Support patch filenames with spaces

Commit:
Fix toolchain deletions in Add toolchain menu

- by pressing ESC (thx kitte888)

Commit:
Sync with Makefile changes in 11739

Commit:
Fix whitespaces

thx Optimum Power

Commit:
Support git binary patches

Commit:
Add csky-linux toolchain to support GX-API builds

Commit:
Fix toolchain deletions in some s3 menus

- by pressing ESC (thx kitte888)
- closes #76

Commit:
Fix unknown function call in Oscam EXTRA menu

- closes #74 (thx @kitte888)

Co-Authored-By: kitte888 <105124981+kitte888@users.noreply.github.com>

Commit:
Fix whitespaces

thx Optimum Power

Commit:
Fix template file not found error related to backing up legacy toolchains

- closes #75 (thx @kitte888)

Commit:
Fix wrong code segments

- use variable instead fixed path names
- remove duplicated function
- closes #77, closes #78 (thx @kitte888)

Commit:
Prepare libdvbcsa static build via USE_STATIC

- Don't create toolchain.cfg native on every start
- Remove empty line during cmd build output
- Fix whitespaces
thx Optimum Power

Commit:
Add USE_LIBDVBCSA switch and libdvbcsa to oscam_name

Commit:
Add gxapi to oscam_name

Commit:
Fix whitespaces

Commit:
Overwrite existing libdvbcsa benchmark tools + libraries zipfile

Commit:
Use oe-mirror as libdvbcsa source

- source is already patched
- remove benchmark tool option

Commit:
Fix some native build error messages

WXbet committed:
Delete support/toolchains.cfg/native

Commit:
Rework libdvbcsa libraries integration options

- add autodection based on toolchain capabilities
- 32/64 versions as option

Commit:
Update libdvbcsa libraries integration options

- display autodection batch size
- warking version compare for autodection

Commit:
Remove force libdvbcsa static build

- to build it static use flag STATIC_LIBDVBCSA
- introduce build flags STATIC_LIBCRYPTO, STATIC_SSL, STATIC_LIBUSB, STATIC_PCSC, STATIC_LIBDVBCSA to enable selective static build
- add -mixed to oscam filename when not all libraries are linked static

Commit:
Fix libdvbcsa libraries integration options

- ensure tokens are properly replaced
- remove some eval commands

Commit:
Update libdvbcsa libraries integration options

- dont create unnecessary symlinks

Commit:
Update output format of tccheck command

Commit:
Remove WITH_UTF8 option

follow https://trac.streamboard.tv/oscam/changeset/11754

Commit:
Fix libdvbcsa neon autodetection for aarch64

and set default to uint64 for this architecture

Commit:
Fix native libdvbcsa static build

Commit:
Introduce generic toolchains based on bootlin

- use precompiled ready to use toolchains from https://toolchains.bootlin.com/
- display generic toolchain on top of lists
- aarch64_generic, armv7_generic, mips32el_generic, sh4_generic, powerpc_generic focus the compatiblity with a wide range of devices

Commit:
Update LIB_PCSC, LIB_CCID library configs

- make sure use host $PKG_CONFIG_BIN binary instead of cross toolchain's one for LIB_PCSC and LIB_CCID
- provide PCSC_LIBS and PCSC_CFLAGS for LIB_CCID

Commit:
Revert s3.TUP Add library static build option

- remove the option LIBS_STATIC_BUILD, this is to confusing
- the library integration scripts produce now both static and dynamic libraries
- the oscam build options decide to use one of it

Commit:
Fix toolchain extraction strip components issue

Commit:
s3.TUP Reset config without startup

- exit tcupdate when called with the --reset (-r) option

Commit:
Better patch log output

Commit:
tccheck: improvement output

- detect both static and dynamic libraries in Library config block

WXbet committed:
Support both oscam svn and git repositories

- the repository type is autodetected based on trunkurl
- rename commands svnpatch | svnrestore | svnup to repopatch | reporestore | repoup
- rename option NO_SVN_AUTOUPDATE to NO_REPO_AUTOUPDATE in support\configs\simplebuild.config

WXbet committed:
Improve repository handling

- repository backups are now created in backup_repo directory (svn: REVISION.tar.gz | git: COMMIT.tar.gz)
- switch trunkurl based on environment variable SOURCE
- all symlinks are now created relativ to better support directory movements

WXbet committed:
Fix switch to specific commits with git

to get the correct branch information the commit cames from

WXbet committed:
Update translations it, tr

commit 442aeee8e0d0b08af04f9c14149a090998957ad6
Author: mapi68 <41143572+mapi68@users.noreply.github.com>
Date:   Tue Apr 9 04:48:25 2024 +0200

    Update it

commit b79d7e64c21837b52bb6fa5e21a4f4655795bb7e
Author: mapi68 <41143572+mapi68@users.noreply.github.com>
Date:   Fri Mar 29 04:08:26 2024 +0100

    Update it

commit 55b4f5c2ddaddf856f89d69e871f622098d0f438
Author: mapi68 <41143572+mapi68@users.noreply.github.com>
Date:   Thu Mar 28 21:52:39 2024 +0100

    Update it

Squashed commit of the following:

commit 496c0fa0cd92e674115a6763db7af0814af636f5
Author: audi06_19 <audi06_19@hotmail.com>
Date:   Wed Mar 27 23:57:53 2024 +0300

    Update tr

WXbet committed:
Update README.md

thx @Optimum Power

WXbet committed:
Fix different revision dialog

- set 0 as default value
- an empty value or ESC closes the dialog without any action

WXbet committed:
Fix toolchain extract

avoid tar "Invalid number of elements" error

WXbet committed:
New upx default --best

WXbet committed:
Fix removed translation tags from languange files

- update german translation (thx Optimum Power)

WXbet committed:
Remove oscam-emu patch support

- oscam-emu patch is deprecated, so switch to use oscam-patched repo checkout
- move trunkurl variable from s3 to support/config/urls
- options enable_emu + disable_emu to switch between trunkurl and emuurl
- option update_emu removed
- show EMU or SOURCE hint in menus when enabled/used

WXbet committed:
Fix toolchain.cfg mca

thx Optimum Power

WXbet committed:
Introduce VERSIONCOUNTER based on rev count

remove deprecated variable (thx Optimum Power)

WXbet committed:
Support git checkout by sha, tag and branch name

- update git branch detection

WXbet committed:
Fix patch flag file

thx Optimum Power

WXbet committed:
Optimize check_url function

- introduce option S3_URL_CHECK to completely disable this check (default=1)
- introduce option PATCH_WEBIF to disable oscam webif patching with listing applied patches (default=1)

WXbet committed:
Fix repoup command

- switch to master branch before pulling changes
- silence git output for repoup command

WXbet committed:
Introduce version option

- ./s3 version displays the current version of simplebuild3
- non-git installations displays a calculated VERSIONCOUNTER with _non-git suffix
- reuse the version function on several places

WXbet committed:
s3.TUP: support OpenSSL 3.3.0

WXbet committed:
Fix branch name and patch flag filename

WXbet committed:
Show revision, commit and branch name on main menu

WXbet committed:
Fix VERSIONCOUNTER prerequisite bc

WXbet committed:
Adjust revision, commit and branch name display on main menu

WXbet committed:
Adapt oscam versioning changes

WXbet committed:
Add extra to oscam_name

WXbet committed:
Whitespace cleanup

thx Optimum Power

WXbet committed:
Remove non existing function call

thx cepheus62

WXbet committed:
s3.TUP Silence architecture missmatch warning

for build system host x86_64 and cross toolchain host i386

WXbet committed:
s3.TUP Backup toolchain: reuse _toolchainfilename from config at legacy toolchains

WXbet committed:
s3.TUP Fix pcsclite 2.2.0 build

- fixed download url for version 1.9.5 as there are wrong download links on https://pcsclite.apdu.fr/files/
- bootstrap call and package autoconf-archive are now mandatory
- pcsc-wirecheck-gen results in "Exec format error" on cross builds, so the file pcsc-wirecheck-dist.c is not created and causes build errors -> so remove 'pcsc_assert_wire_constants ()' call  from "pcsc-wirecheck-main.c" via sed

WXbet committed:
Introduce libdvbcsa native installer

WXbet committed:
Fix autoconf-archive installed detection

WXbet committed:
tccheck: Fix zlib detection for native

WXbet committed:
s3.TUP Support recently introduced options in _create_toolchaincfg function

- add extra_c, _extract_strip variables
- handle legacy toolchains correctly for_description, _toolchainfilename variables
- dynamic counting lines for _tc_infolines variable based on _tc_info variable contents

WXbet committed:
Update libdvbcsa native installer

- new s3 option to configure libdvbcsa installer INSTALL_NATIVE_LIBDVBCSA=0|1 (default=1)
- remove directory not found errors
- add dvbcsa.h header to syscheck

WXbet committed:
More compatible libdvbcsa +upx native installer

- syscheck: install correct dependent packages, fix ldconfig call, add simple upx precompiled binary installer
- check_pkg: move some packages to syscheck, remove subversion package
- debian-ubuntu: switch to sudo for debian, add upx installer call, support for Debian 10|11|12 + Ubuntu 20|22|24.04
- manjaro.install: add upx installer call
- redhat_centos_fedora.install: add upx installer call
- suse.install: add upx installer call

WXbet committed:
Introduce new git based online updater

WXbet committed:
Improve git based online updater

- log update process to update_me.log
- cleanup working copy (checkout, stash, delete)
- use --no-pager option for git commands to prevent  (e.g. git log) from paging
- remove update related functions from s3.TUP
- move _log function to _misc module

WXbet committed:
Update: Do not switch to master branch by default

as this will prevent updating s3 cloned from another branch

- remove some left comments

WXbet committed:
s3.TUP Fix pcsclite 2.2.1 build

- fix version number in configure.ac as it is not properly set since 2.2.1

WXbet committed:
s3.TUP Fix _sysroot path creation in _create_toolchaincfg function

- support for symlinked support/toolchain folder

WXbet committed:
Update: Handle forced pushes on remote repository

WXbet committed:
s3.TUP Freetz-NG toolchain template updates

freetz toolchains pack15

- remove all other fritz toolchains as they are outdated

Co-Authored-By: MasterRoCcO <47879729+MasterRoCcO@users.noreply.github.com>

WXbet committed:
Update package installer

reactivate and improve support for non-sudo installation via su

WXbet committed:
Improve libdvbcsa neon + altivec autodetection

WXbet committed:
tccheck: Add gdb and binutils (ld) version detection

WXbet committed:
s3.TUP Fix CCID 1.6.0 build

- bootstrap call and package autoconf-archive are now mandatory

WXbet committed:
Fix upx native installer

add arm detection

WXbet committed:
Fix trailing whitespaces

WXbet committed:
Improve log output during cmd and menu build

WXbet committed:
Follow gitlab upx changes

- remove upx compression functions

WXbet committed:
Speed up git cloning

- new config option GIT_CLONE_DEPTH (default "1") to active shallow cloned git repos (much faster) in simplebuild.config
- to increase the cloned git repo history entries set GIT_CLONE_DEPTH to the desired count
- to work with slower full cloned git repos set GIT_SHALLOW_CLONE="0"

WXbet committed:
Some menu fixes

- update descriptions for OSCAM_EXTRA, BUILD_EXTRA, SELECT_STAPI in configuation menu
- cleanup and sort entries in Oscam EXTRA menu
- cleanup and reorder entries in Build EXTRA menu

WXbet committed:
Update symlink creation for profile native

- new symlinks for objcopy and objdump

WXbet committed:
Update oscam_name generation

- move -upx after -extra, -mixed, -static
- detect oscam binary signing option and add -signed at the end

WXbet committed:
Improve log output during cmd and menu build

WXbet committed:
Add option to use own certificates for oscam signing

WXbet committed:
s3.TUP Restore Android toolchain functionality

WXbet committed:
s3.TUP SSL config updates for Android

WXbet committed:
Fix git binary patch support

WXbet committed:
Fix static build warning

warning: -jN forced in submake: resetting jobserver mode

WXbet committed:
s3.TUP Fix libusb + pcsclite + libdvbcsa build for csky

WXbet committed:
s3.TUP Create zip archive containing each integrated library

- zip archive TOOLCHAIN-NAME-libs.zip is located in binaries folder
- zip archive is automatically updated when a library is integrated
- pcsc_scan build has been added to PCSC-Lite
- libusb 0.1.12 download url changed

WXbet committed:
s3.TUP Remove libatomic dependency from OpenSSL

- add PCSC_LIBS and PCSC_CFLAGS parameter to pcsc-tools configure call

WXbet committed:
Do not use deprecated config.sh option

- remove unused EMU function
- fix maxdepth for find command

WXbet committed:
Remove enable_emu + disable_emu functions

as oscam_patched repository is history.
Simple oscam source url switching is still possible via oscamsource.url file, see https://github.com/gorgone/s3_releases/blob/master/s3#L141 ...

WXbet committed:
Fix handling of value 0 in checkout menu

WXbet committed:
s3.TUP: support OpenSSL 3.4.0

WXbet committed:
Add oscamsource.url.template

- describe usage of source url switching
- display currently used repository in logo

WXbet committed:
Display wc url deviation

WXbet committed:
More use of repo identifier from oscamsource.url

- add identifier to working copy folder name
- add identifier to backup file names
- display currently used working copy folder name in logo
-

WXbet committed:
New option repoclean

to remove working copy and all backup files including identifier support

WXbet committed:
Update bg translation (thx @tedi58)

WXbet committed:
Try to support more OpenSSH options

- first, try to copy without specific option (default protocol in OpenSSH before version 9 is SCP, in OpenSSH 9+ it is SFTP)
- second, try to copy with -s parameter (force SFTP protocol, OpenSSH before version 9)
- third, try to copy with -O parameter (force SCP protocol, OpenSSH 9+)

WXbet committed:
Rework upload command

- beautify commandline output
- enable stop_target
- add missing scp option
- more verbose error messages

WXbet committed:
Remove duplicated code for static build options

-

WXbet committed:
Beautifying s3 help output

- sort options
- display missing USE_SSL use_var

WXbet committed:
Optimize check_url function

- S3_URL_CHECK option new default=3 (connection timeout in seconds)
- S3_URL_CHECK option=0 to disable url check

WXbet committed:
Restore correct handling of USE-variables during gui build

- revert parts of https://github.com/gorgone/s3_releases/commit/344b277dd9768f2d74ef69fa8bdb220127922902

WXbet committed:
s3.TUP Update OpenSSL build parameter for Android

WXbet committed:
Filter option for repopatch command

WXbet committed:
upload_cam: Allow custom command in stop_target variable

- supported values: "y" or "n" or "my custom command1; my custom command2;"
- add missing variable to upload.example.ssh

WXbet committed:
Introduce force option for repopatch command

e.g.
./s3 repopatch -f
./s3 repopatch --force
./s3 repopatch "filter*" -f
./s3 repopatch "filter*"--force

WXbet committed:
Fix unknown parameter error during subversion checkout

WXbet committed:
Rework patch style detection and precise output

WXbet committed:
Backward compatibility for getting oscam revision

WXbet committed:
upload_cam: Allow older HostKeyAlgorithms by default

WXbet committed:
upx_native_installer - fix download url for arch arm

WXbet committed:
s3.TUP Add OpenSSL 3.5.0 (alpha)

- set LIBS_LIST_BETA="1"; in support\configs\plugin_update_toolchain.config to make it usable

WXbet committed:
upx_native_installer - exclude version 5.0.0 (too buggy)

WXbet committed:
s3.TUP Switch to OpenSSL 3.5.0 final

WXbet committed:
s3.TUP Add OpenSSL 3.6.0 (alpha)

WXbet committed:
s3.TUP Switch to OpenSSL 3.6.0 (release)

WXbet committed:
build: update cc_warn flags

- use EXTRA_CC_WARN to append CC_WARN in Makefile instead of overwriting it

WXbet committed:
cmd_build: create lastbuild.cmd for reproducability

- improve logfile date
- speed up oscam module (en-|dis-)abling
-

WXbet committed:
cmd_build: avoid false positive erros

WXbet committed:
build: show optional git submodule output

WXbet committed:
cmd_build: create lastbuild.cmd optionally (default=off)

WXbet committed:
cmd_build: restore enable/disable logic

accidently changed in https://github.com/gorgone/s3_releases/commit/ff234b8c6b5400a850461a77ba8d786ae81f42d0

WXbet committed:
generate_oscam_name: add security backend if oscam supports it

WXbet committed:
build: fix USE_FLAGS export for hardware-specific cardreader selection

Export active USE_* flags to environment before build so that
config.sh can properly detect hardware-specific options when
generating config.mak.

Without this export, config.sh could not determine which internal
cardreader variant to enable (e.g. CARDREADER_INTERNAL_AZBOX,
CARDREADER_INTERNAL_COOLAPI, etc.) because the USE_* flags were
only passed to make but not available during config.mak generation.

WXbet committed:
build: show WIKI build output

WXbet committed:
s3.TUP Fix zlib download urls

WXbet committed:
s3.TUP improve jq commands

and add -lrt to libusb

WXbet committed:
set_buildtype: prefer sysroot/usr/lib path for static library detection

Add _find_static_lib() helper function that prioritizes libraries
in */usr/lib/* paths over other locations. This fixes an issue where
toolchains with duplicate libraries (e.g. solo4k) would incorrectly
use outdated libraries from the wrong path, causing linker errors
like "undefined reference to clock_gettime".

Affected libraries: libcrypto, libssl, libusb, libpcsclite, libdvbcsa

WXbet committed:
s3.TUP introduce some backup urls

WXbet committed:
s3.TUP fix introduce some backup urls

WXbet committed:
Conditional download with caching in _dl() function

- Add filename detection from URL using regex for known archive extensions
- Check server reachability before downloading when local file exists
- Use wget -N for conditional download (only if newer)
- Keep existing local file silently when server is unreachable
- Fall back to --content-disposition for dynamic URLs without filename
- Fix _sys_check to skip directories in source loop

WXbet committed:
Introduce support for OE-SDK toolchains

OE/Yocto SDK toolchains have gcc compiled with --sysroot=/not/exist
and rely on wrapper scripts or environment variables to set the
correct sysroot at runtime. This breaks s3's direct gcc invocations
for capability tests (C11, neon, altivec), sysroot detection, and
the oscam build (TOOLCHAIN_INC_DIR).

_toolchain:
- Fallback to _sysroot from config when gcc -print-sysroot returns
  an invalid directory
- Pass --sysroot and extra_cc flags to compiler capability tests
  (c11, neon, altivec) so they work without a valid built-in sysroot

_plugin_update_toolchain:
- Same sysroot fallback in _get_toolchain_properties
- Suppress realpath error on invalid include path detection
- Source toolchain config in _integrate_libs so $_sysroot and
  $extra_cc are available for capability checks during tcupdate
- Add reverse version match: detect installed base versions (e.g.
  1.1.0) as matching when expected version has a suffix (e.g.
  1.1.0-128-neon), treating them as equal

_cmd_build / _gui_build:
- Auto-inject --sysroot into extra_cc when gcc's built-in sysroot
  is invalid, placed after the USE_EXTRA unset block
- Override CC= in make invocation so oscam's Makefile resolves
  TOOLCHAIN_INC_DIR correctly via $(CC) -Wp,-v

All changes are guarded by [ ! -d "$sysroot" ] checks so existing
crosstool toolchains with valid built-in sysroots are unaffected.

WXbet committed:
Fix sysroot auto-inject breaking native toolchain builds

The sysroot fallback incorrectly triggered for native toolchains
where gcc -print-sysroot returns an empty string (normal behavior
for system compilers). This caused --sysroot to be set to an
invalid path like /opt/s3/support/toolchains/native/usr/include.

Add [ -n "$..." ] guard to all sysroot fallback checks so they
only trigger when gcc returns a non-empty but invalid sysroot
path (like /not/exist from OE-SDK toolchains), not when it
returns empty (native/system compiler).

WXbet committed:
Add Open Embedded SDK (OESDK) toolchain builder

Integrate OE/Yocto SDK toolchain creation into s3's existing
toolchain build system alongside CTNG, FNG, and ANDK builders.

OESDK toolchains are built by cloning a build-environment repo
(oe-alliance, openpli, etc.), running bitbake meta-toolchain with
configurable SDK extras, and installing the resulting SDK into the
s3 toolchain directory with a bin/ symlink.

Config template (plugin_update_toolchain.config.template):
- OESDK_BUILD_tasks: resolve OEM, detect build directory, source
  environment, bitbake meta-toolchain, install SDK, create symlink
- OESDK_SETUP_tasks: clone repo
- OESDK_CONFIG_tasks: editor
- Build directory auto-detected from TOPDIR in conf files

OEM resolution (3-stage fallback):
1. OESDK_OEM from template if set
2. make list (MACHINEBUILD -> MACHINE mapping, new-style Makefile)
3. Parse ifeq blocks from Makefile (old-style Makefile)

Plugin (_plugin_update_toolchain):
- oesdksrcdir resolved dynamically from OESDK_REPO_LOCATION
  (relative to crosstool dir, or absolute path for existing envs)
- OESDK type detection in _get_template_type() via OESDK_ prefix
- @OESDKSOURCE@ token in _replace_tokens()
- _oesdk_setup() function for repo clone
- OESDK in _check_crosstool_setup() (only when oesdksrcdir set)
- OESDK cases in _create_tc(): template prep, build tasks, error
  detection, post-build (target/sysroot/tune_ccargs/dynamic linker
  extracted from SDK environment-setup script)
- OESDK in _get_template_properties() for arch/gcc/glibc/linux
- OESDK in _get_toolchain_date() via oe-sdk.log.bz2
- Skip library auto-integrate for OESDK (libs already in sysroot)
- Inject --sysroot and tune_ccargs into cflags/ldflags for
  library builds via _get_toolchain_properties()
- Auto-detect dynamic linker from sysroot for extra_ld
- tune_ccargs routed to extra_cc (pos 12) via separate
  _create_toolchaincfg call for OESDK

Templates (per-template configuration):
- OESDK_MACHINE, OESDK_OEM, OESDK_DISTRO for target selection
- OESDK_REPO_URL, OESDK_REPO_BRANCH for build environment source
- OESDK_REPO_LOCATION for shared/custom build environment paths
  (supports variable expansion: oe-sdk_${OESDK_DISTRO}_${OESDK_REPO_BRANCH})
- OESDK_SDK_EXTRAS for additional dev packages in SDK
- 9 example templates: pulse4kmini/sf8008/vusolo2 x
  openatv 8.0/openatv 7.6/openpli 10.0

WXbet committed:
Add post-build QEMU smoke test for cross-compiled binaries

Run a QEMU user-mode emulation test after successful builds to
verify the binary starts correctly without needing target hardware.

The test starts the built binary under qemu-{arch} with the
toolchain's sysroot, monitors output for an expected string
(default: "Server running"), and reports PASS/FAIL. Live output
streaming shows oscam's startup log during the test.

Configuration via simplebuild.config:
- POSTBUILD_QEMU_TEST: enable/disable (default: 0)
- POSTBUILD_QEMU_TIMEOUT: max wait in seconds (default: 5)
- POSTBUILD_QEMU_PARAM: extra parameters for the binary
- POSTBUILD_QEMU_EXPECT: string to match in output

Implementation:
- _postbuild_qemu_test() in _toolchain (shared function)
- Called from both _cmd_build (CLI) and _gui_build (dialog)
- QEMU arch auto-detected via ${CROSS}gcc -dumpmachine
- Test skipped if qemu binary not installed or build had errors
- QEMU output streamed live and appended to build log
- QEMU command recorded in build cmd file (CREATE_BUILDCMD_FILE)

Prerequisite: install qemu-user-binfmt
  sudo apt install qemu-user-binfmt

WXbet committed:
Only install the toolchain builder required by the selected template

Previously _check_crosstool_setup() was called without parameters,
causing all builders (crosstool-NG, Freetz-NG, Android-NDK, OE-SDK)
to be installed even when only one was needed.

Now the template type is passed to _check_crosstool_setup() so only
the relevant builder is set up:
- CTNG template -> only crosstool-NG
- FNG template -> only Freetz-NG
- ANDK template -> only Android-NDK
- OESDK template -> only OE-SDK clone

Applies to both the build loop and the template editor.

WXbet committed:
Support opendreambox build environment for OESDK toolchains

opendreambox uses a different build system layout than OE-Alliance
and OpenPLi, requiring several adaptations:

Build environment detection:
- Source bitbake.env (opendreambox) when oe-init-build-env fails
  or env.source doesn't exist, with oe-init-build-env as fallback
- Fallback PATH injection if bitbake not found after sourcing
  (handles moved/symlinked repos with stale paths in bitbake.env)
- Export MACHINE after environment setup (opendreambox env.source
  does not set MACHINE, only exports the variable name)
- Dynamic SDK installer discovery via find instead of hardcoded
  ./tmp/deploy/sdk/ (opendreambox uses tmp-glibc/)
- Auto-detect _append vs :append syntax based on BitBake version
  (krogoth uses _append, scarthgap+ uses :append)
- Exit on bitbake or SDK installer failure (|| exit 1)

Toolchain config generation:
- Fallback to extract TUNE_CCARGS from CC= export line when
  OECORE_TUNE_CCARGS is not available (older OE SDKs like krogoth)

Template:
- Add dm920_opendreambox_2.5 template sample for Dreambox DM920
  using opendreambox krogoth branch

WXbet committed:
Strip OESDK toolchains and fix build error handling

Strip unnecessary files from the SDK host sysroot after install
to reduce toolchain size by ~75% (1.7GB -> 445MB). Removes
locales, QEMU firmware, Python, cmake, autoconf, automake,
libtool, gettext, documentation, and unused native binaries.

Add set -o pipefail to OESDK build tasks so bitbake failures
are correctly detected through tee pipes (exit 1 triggers on
bitbake error instead of tee's exit 0).

WXbet committed:
Fix build errors with OESDK toolchains

Two issues when building libraries (tcupdate) with OESDK toolchains:

@CC@ token: Quote and append --sysroot and tune flags for OESDK
toolchains so that all library configure scripts receive the correct sysroot
and float-abi flags via $CC. Without this, configure ignores $CFLAGS
and compiles with wrong defaults.

PKG_CONFIG_PATH: Use tr to join multiple pkgconfig directory paths
with colons instead of newlines. OESDK sysroots contain both
usr/lib/pkgconfig and usr/share/pkgconfig, causing find to return
newline-separated paths that break configure scripts.

WXbet committed:
Normalize whitespace in OESDK extra_cc flags

Trim leading spaces and collapse multiple spaces in tune_ccargs
extracted from SDK environment-setup scripts.

WXbet committed:
Add static library packages to OESDK SDK extras

OE/Yocto splits static libraries (.a) into separate -staticdev
packages that are not installed by default with -dev packages.
Without them, oscam's static linking options (STATIC_LIBCRYPTO,
STATIC_SSL, STATIC_LIBUSB, STATIC_PCSC, STATIC_LIBDVBCSA) fail
at link time with undefined references.

Add -staticdev packages for all libraries in OESDK_SDK_EXTRAS:
openssl, libusb1, libdvbcsa, pcsc-lite, curl.

WXbet committed:
Bump version to 2026.03

WXbet committed:
Fix @CC@ token quoting for OESDK toolchains in library builds

OESDK toolchains inject tune flags and --sysroot into @CC@, making
it contain spaces (e.g. "gcc -march=armv7ve --sysroot=/path").
This broke two things:

1. export CC=@CC@ without quotes caused the shell to interpret
   flags as separate commands instead of part of CC value

2. [[ @CC@ =~ "csky" ]] with quoted @CC@ produced double-quotes
   and syntax errors in bash conditional expressions

Fix: Use two-pass sed replacement in _replace_tokens():
- First pass: replace "export CC=@CC@" with quoted value
- Second pass: replace remaining @CC@ occurrences unquoted

This ensures CC exports work with spaces while regex tests
in [[ ]] expressions remain valid. Only affects OESDK toolchains
where @CC@ contains tune flags; standard toolchains are unchanged.

WXbet committed:
Use patchelf for portable OESDK toolchains

Replace relocate_sdk.py with patchelf to set the system dynamic
loader (/lib64/ld-linux-x86-64.so.2) on all SDK host binaries.
This makes OESDK toolchains portable — they can be copied to any
x86_64 Linux host and work immediately without path adjustments.

The system loader path is auto-detected via file -L on the host
bash binary, so it works on any distribution.

patchelf runs after SDK stripping (fewer binaries to patch) and
is added as a dependency in _plugin_update_toolchain syscheck.

Also adds dreamone_opendreambox_2.6 template (aarch64, pyro branch).

WXbet committed:
Fix multilib filtering and bitness detection in OESDK post-build

- Filter out lib32/multilib environment-setup scripts in post-build
  case to get correct target, sysroot, and tune_ccargs for aarch64
  machines like Dream One
- Same multilib filter in _get_template_properties for consistent
  toolchain info display
- Auto-detect bitness from arch (aarch64/x86_64/mips64 = 64-bit)
  instead of hardcoding 32-bit
- Fix _tc_infolines: use echo -e to interpret \n so wc -l returns
  correct line count (affects all toolchain types)

WXbet committed:
Auto-relocate OESDK toolchain interpreter on first use

Using patchelf to set the system loader at build time breaks on
hosts where the SDK glibc is incompatible with the system glibc
(e.g. SDK glibc 2.23 on a host with glibc 2.27 and different
CPU features).

Instead, keep the SDK-internal loader during build. On first use,
_oesdk_relocate() checks if the interpreter path in the gcc binary
is valid. If not (e.g. after copying/restoring the toolchain to a
different path), patchelf sets the interpreter to the SDK-internal
loader at its new location. This keeps the SDK self-contained and
works regardless of host glibc version or CPU architecture.

- Remove patchelf from OESDK_BUILD_tasks
- Add _oesdk_relocate() function in _toolchain that auto-detects
  invalid interpreter paths via readelf and patches them
- Call _oesdk_relocate from _check_toolchain so it triggers
  automatically before any build, tccheck, or tcupdate

WXbet committed:
Auto-install qemu and patchelf via syscheck when needed

QEMU: When the post-build smoke test is enabled but the required
qemu binary is not found, trigger syscheck to automatically
install the appropriate package:
- Debian/Ubuntu: qemu-user-binfmt
- Fedora/RHEL: qemu-user-static
- Arch/Manjaro: qemu-user-static
- openSUSE: qemu-linux-user

The qemu-* wildcard pattern in the OS installers matches any
architecture variant (qemu-arm, qemu-aarch64, qemu-mipsel, etc.)

patchelf: Move from global _check_pkg dependency to on-demand
install in _oesdk_relocate. Only installed when an OESDK toolchain
actually needs interpreter relocation, not for every tcupdate.

WXbet committed:
Fix OESDK toolchain relocate using relocate_sdk.py

Replace patchelf-only approach with relocate_sdk.py which correctly
patches both the ELF interpreter AND the loader's internal library
search paths (.sysdirs, .ldsocache, .gccrelocprefix sections).
patchelf alone left the loader's hardcoded paths unchanged, causing
"cannot open shared object" errors after toolchain rename/move.

_oesdk_relocate() now:
1. Reads interpreter path from a real gcc binary (skips empty stubs)
2. Checks if the loader at that path exists — if yes, return early
3. Derives old prefix from interpreter path (before /sysroots/)
4. Updates old_prefix in relocate_sdk.py via sed
5. Runs relocate_sdk.py to patch all host binaries and the loader
6. Updates toolchains.cfg with new toolchain name
7. Sleeps 2s so the relocate message is visible before screen clear

Also:
- Copy relocate_sdk.py into toolchain during build (OESDK_BUILD_tasks)
- Create lib/usr/etc symlinks in toolchain root during build
- Call _oesdk_relocate from _pre_build before sourcing toolchains.cfg
  to ensure cfg has correct _toolchainname before it is loaded
- Remove pcsc-lite-staticdev from OpenPLi templates (not available)

WXbet committed:
Fix multi-machine SDK builds and QEMU syscall noise

When multiple machines share the same OE build environment, the
SDK installer and environment-setup files from previous builds
remain in the deploy directory. This caused wrong SDK installers
and tune flags to be used for subsequent machines.

Config template:
- Select SDK installer by machine name (-${OESDK_OEM}-) instead
  of taking the first .sh found, with fallback for compatibility
- Clean toolchain directory before SDK install to prevent stale
  environment-setup files from previous machines mixing in
  (only deletes if sysroots/ exists as safety guard)

QEMU smoke test:
- Filter "qemu: Unsupported syscall" warnings from output display
  (harmless noise from older QEMU versions on newer kernels)

WXbet committed:
Rename OE-Alliance templates, add static lib support, fix tc_infolines

- Rename templates from *_openatv_* to *_oealliance_* to reflect
  they use the OE-Alliance build environment (not distro-specific)
- Add staticdev packages to all OE-Alliance templates for static
  linking support (openssl, libusb1, libdvbcsa, curl)
- Enable static library builds per-recipe via DISABLE_STATIC
  in sdk-extras.conf (only for packages with -staticdev requested)
- Fix _tc_infolines calculation: use echo without -e flag so
  escaped \n in the info string are not expanded, giving correct
  line count

WXbet committed:
Fix make init to use OESDK_MACHINE instead of OESDK_OEM

OESDK_OEM is derived after init and may differ from the actual
MACHINE value needed by the build system. Use OESDK_MACHINE
directly to ensure correct build directory setup.

WXbet committed:
Pass default_use as parameter to _create_toolchaincfg

Make default_use configurable via 18th parameter instead of
hardcoding USE_LIBCRYPTO. OESDK toolchains get USE_LIBCRYPTO
USE_EXTRA, others get USE_LIBCRYPTO. Backup preserves the
original default_use from the sourced config.

WXbet committed:
Fix OEM lookup awk pattern surviving eval escaping

The awk regex for MACHINEBUILD lookup broke after passing through
single-quote template, array expansion and eval. Simplify the
pattern to avoid backslash escaping issues.

WXbet committed:
Optimize toolchain menu, add OE-SDK build script, machine search and OESDK fixes

Build script (build-openembedded-toolchains.sh):
- Add Open-Embedded SDK Toolchain Builder for batch building deduplicated
  toolchains grouped by DEFAULTTUNE
- Support OE-Alliance, OpenDreambox, OpenPLi and other OE-based build envs
- Generate crosstool templates with OESDK_MACHINES (all MACHINEBUILDs)
- Include .config in tar.xz for version tracking after download
- Alphabetically sorted machine lists across all build environments
- Auto-detect repo URL and branch from build env git info

Plugin optimizations (_plugin_update_toolchain):
- Reduce template menu load time ~80%: single-pass awk for header parsing,
  CTNG/ANDK property extraction, and template type detection
- Fix _get_template_type awk END block bug (exit triggers END in awk)
- Fix OESDK templates: show Download/Build hint instead of Setup
- Fix printf '%q' escaping that produced backslash artifacts in menu
- Fix OESDK template editing when build env directory doesn't exist
- Add FNG prerequisites install to setup tasks

Toolchain menu (_toolchain):
- Add SEARCH button to Add Toolchain and Toolchain main menu
- Substring search across _machines and toolchain names
- Show active filter in menu title, BACK always returns to main menu

Translations:
- Add txt_search, txt_search_machine for all languages
- Add txt_s3tup_msg_gtp_info_oesdk for OESDK-specific hint text

New templates and configs for OE-Alliance, OpenDreambox, OpenPLi toolchains.

WXbet committed:
Fix SSL library detection, optimize tcupdate performance

- Fix SSL version check: use Major.Minor match for OpenSSL 3.x series
  (3.6 matches 3.6.0 and 3.6.1), keep Major.Minor.Patch for 1.x/0.x
  to avoid collisions (1.1.1 vs 1.1.0). Config version 97 -> 98.
- Resolve ANDK_PKG_URL/SHA1 curl calls at config creation time instead
  of every source, reducing config load from 1.8s to 0.017s.
- Add pre-check in _get_toolchain_libs: skip expensive _replace_tokens
  and _compare_version when .pc filename not in check string. Reduces
  lib detection from 51s to 1.25s for OE-SDK toolchains (33 .pc files).
- Cache _check_compiler_capability results per toolchain per session.
- Skip _replace_tokens sed pipeline when input has no @ tokens.
- Reset CROSS_SYSROOT in _gui_build and _cmd_build to prevent stale
  sysroot from previous toolchain leaking into next build.

WXbet committed:
Revert "build: fix USE_FLAGS export for hardware-specific cardreader selection"

This reverts commit 3c04e238d5451c021390a8efce07e4f49c2421c0.

WXbet committed:
Fix -ssl suffix detection in binary name

oscam removed WITH_SSL as a config.sh option; it now exists only as a
-DWITH_SSL=1 compile flag set by the Makefile's prepare_use_flags macro.
config.sh --enabled WITH_SSL therefore always returns N, dropping the
-ssl suffix from the binary name.

Detect USE_SSL directly from USE_vars[USE_SSL] instead.
